### PR TITLE
Sockets - Maria 

### PR DESCRIPTION
**Pull request by MariaWissler**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | For me was more about the understanding of what i needed to do, loops, declare variables and make a "hash"
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? |  I think the only thing  I added to all my searchs was javascript 
What was a challenge you were able to overcome on this assignment? | many !, definetely feel less  scare about the syntax on jaa script and how to create the methods~functions 
What has been interesting and positive about learning a new programming language? | js seems fun, like it lets you do a lot of things for ruby seemed so friendly and more orderly, maybe im too new to js 
What is something to focus on your learnings in JavaScript in the next week? |  practice using loops, itirations and get more familiar with available methods

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,8 +1,170 @@
...
+    "W", "W", // W : 2
+    "X", // X: 1
+    "Y", "Y", // Y: 2
+    "Z" // Z: 1
+  ];
+
+  let lettersInHand = []
+  for (let i = 0; i < 10; i++) { //for (int i = 0; i < array.length; i++)
+    const letterPool = (letters)//Invoking this function should not change the pool of letters (makes copy)
+    let randomLetters = letterPool.splice(Math.floor(letterPool.length * Math.random()), 1)[0];// splice remove one at index [0]
```
Comment (dHelmgren) [inline src/adagrams.js:39]: using letterPool inside the loop is actually making a copy each time the loop starts, which means that your letters in hand is always drawing from the "full pool", with the initial probabilities getting repeated. I believe it's possible, in other words, for you to draw more than one z in a single hand.

File: src/adagrams.js [hunk 2]
```diff
@@ -1,8 +1,170 @@
...
+  return lettersInHand;
+ },
+
+ usesAvailableLetters(input, lettersInHand) {
+ let inputWord = input.split(""); // split input from user Array ["T", "h", "e"]
+
+ for (let letter of inputWord) { // for every letter in input 
+  let index = lettersInHand.indexOf(letter);// index of where the letter can be found in the array
+  if (index != -1) { //-1 if letter is not present.
+    lettersInHand.splice(index, 1);//removes all the letters if present
```
Comment (dHelmgren) [inline src/adagrams.js:51]: nice!

File: src/adagrams.js [hunk 3]
```diff
@@ -1,8 +1,170 @@
...
+  const valueFour = ["F", "H", "V", "W", "Y"];	     
+  const valueFive = ["K"];	     
+  const valueEight = ["J", "X"];	     
+  const valueTen = ["Q", "Z"];	          
+  let splitWord = word.toUpperCase().split("");   // ["T", "H", "E"] 
+  let wordLength = word.length;
+  let score = 0;     
+  
+  for (let letter of splitWord) {//E  -  score = 6
+    if(valueOne.indexOf(letter)>-1){
```
Comment (dHelmgren) [inline src/adagrams.js:73]: Good approach using a conditional! You probably could refactor this to use an object (a hash) though. :)

File: src/adagrams.js [hunk 4]
```diff
@@ -1,8 +1,170 @@
...
+        }
+      }
+      if(maxRep === 1) {// if one, we return winner 
+        return {word:wordMaxScore, score:maxScore};
+      }
+      else{ 
+        let wordArray = [];
+          for(let item of list){ //lets get those max
+            if(item.score==maxScore){
+              wordArray.push(item);
```
Comment (dHelmgren) [inline src/adagrams.js:134]: Why not make this list as you're checking to see if there is a second max high score?

File: src/adagrams.js [hunk 5]
```diff
@@ -1,8 +1,170 @@
...
+          wordArray.sort((a, b) => (a.length > b.length) ? 1 : -1); //asc de min to max
+          let minLength = wordArray[0].length; //lenght min
+          let maxLength = wordArray[wordArray.length-1].length; //length max
+          let maxRepMinLength = 0;
+          let maxRepMaxLenght = 0;
+          
+          for(let item of wordArray) {// repeats ?
+            if(item.length == minLength){
+              maxRepMinLength = maxRepMinLength + 1;//maxRep++;
+              wordMaxScore = item.word;
```
Comment (dHelmgren) [inline src/adagrams.js:146]: This piece here is kind of verbose. I think once you've found the shortest, you return that, or you return the one whose length == 10

**Comment by dHelmgren** [comment]
# JS Adagrams
## What We're Looking For



Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | Yes
Small commits with meaningful commit messages | no. Especially when working in a new language, it's better to make tiny, constant commits.
**Code Requirements** |
`drawLetters` method | good, see comment re: probability
Uses appropriate data structure to store the letter distribution | yes
All tests for `drawLetters` pass | yes
`usesAvailableLetters` method | yes
All tests for `usesAvailableLetters` pass | yes
`scoreWord` method | yes
Uses appropriate data structure to store the letter scores | yes
All tests for `scoreWord` pass | yes
`highestScoreFrom` method | yes
Appropriately handles edge cases for tie-breaking logic | yes, but see comment
All tests for `highestScoreFrom` pass | yes
**Overall** | Great work on this project! In this project you’ve taken some interesting logic and worked it into JavaScript syntax. I’m adding a few comments on places that this could be better and minor suggestions for improvement, but overall, you’ve done a great job of practicing syntax.
